crates/executor/src/executor.rs

```diff
@@ -17,7 +18,157 @@ use revm::{
+        self.evm.env.cfg.spec_id = spec_id;
+        self.evm.env.cfg.perf_all_precompiles_have_balance = false;
+        self.evm.env.cfg.perf_analyse_created_bytecodes = AnalysisKind::Raw;
+
+        revm_wrap::fill_block_env(&mut self.evm.env.block, header, spec_id >= SpecId::MERGE);
+    }
+}
+
+impl<'a, DB> BlockExecutor for Executor<'a, DB>
+where
+    DB: StateProvider,
+{
+    fn execute(
+        &mut self,
+        block: &Block,
+        signers: Option<Vec<Address>>,
+    ) -> Result<ExecutionResult, Error> {
+        let Block { header, body, ommers } = block;
+        let signers = self.recover_signers(body, signers)?;
+
+        self.init_block_env(header);
+
+        let mut cumulative_gas_used = 0;
+        // output of verification
+        let mut changesets = Vec::with_capacity(body.len());
+
+        for (transaction, signer) in body.iter().zip(signers.into_iter()) {
+            // The sum of the transaction’s gas limit, Tg, and the gas utilised in this block prior,
+            // must be no greater than the block’s gasLimit.
+            let block_available_gas = header.gas_limit - cumulative_gas_used;
+            if transaction.gas_limit() > block_available_gas {
+                return Err(Error::TransactionGasLimitMoreThenAvailableBlockGas {
+                    transaction_gas_limit: transaction.gas_limit(),
+                    block_available_gas,
+                })
+            }
+
+            // Fill revm structure.
+            revm_wrap::fill_tx_env(&mut self.evm.env.tx, transaction, signer);
+
+            // Execute transaction.
+            let out = self.evm.transact();
+
+            // Useful for debugging
+            // let out = evm.inspect(revm::inspectors::CustomPrintTracer::default());
+            // tracing::trace!(target:"evm","Executing transaction {:?}, \n:{out:?}: {:?}
+            // \nENV:{:?}",transaction.hash(),transaction,evm.env);
+
+            let (revm::ExecutionResult { exit_reason, gas_used, logs, .. }, state) = out;
+
+            // Fatal internal error.
+            if exit_reason == revm::Return::FatalExternalError {
+                return Err(Error::ExecutionFatalError)
+            }
+
+            // Success flag was added in `EIP-658: Embedding transaction status code in receipts`.
+            // TODO for verification (exit_reason): some error should return EVM error as the block
+            // with that transaction can have consensus error that would make block
+            // invalid.
+            let is_success = match exit_reason {
+                revm::return_ok!() => true,
+                revm::return_revert!() => false,
+                _ => false,
+                //e => return Err(Error::EVMError { error_code: e as u32 }),
+            };
+
+            // Add spend gas.
+            cumulative_gas_used += gas_used;
+
+            // Transform logs to reth format.
+            let logs: Vec<Log> = logs
```

Comment:
Nit: we should impl `From<reth::Log> for Log` somewhere instead so we can just do `logs.into()`

Reply:
Definitely agree but I realized why the original auther wrote this way. We should impl `impl From<revm::Log> for reth_primitives::Log`. It could be only defined in `reth_premitives` but we don't want that `reth_premitives` depends on `revm`. 

Altenative: I'll write a utility function in `reth_wrap.rs`. There is already this kind of function. e.g, `to_reth_acc()`.

Reply:
follow it up w/ ea6e078d7347b485faa3d6344cdd33e8c20f8a44